errno 34 (floppy error) when calling C.nn_socket

I have this weird error. errno is 34 which doesn't make sense. I tested with an older version of nanomsg but I still have the problem. I'm wondering if I got this since I upgrader to go 1.1.2.

The last time I edited nanomsg and use printf to show errno, it wasn't 34 even if I got it when calling C.nn_socket.

I'm going to try with go 1.1.1.

I'm on windows/amd64.

```
C:\dev\go-nanomsg>go test
--- FAIL: TestError (0.01 seconds)
        error_test.go:15: Le lecteur ne contient pas la bonne disquette.
                Insérez %2 (numéro de série de volume : %3) dans le lecteur %1.
FAIL
exit status 1
FAIL    _/C_/dev/go-nanomsg     0.171s
```

I build nanomsg using VC 2012 express and I use mingw64 with Go. Could this be the source of my problem?

--- a/nanomsg.go
+++ b/nanomsg.go
@@ -41,7 +41,7 @@ type Socket struct {
 // Create a socket.
 func NewSocket(domain Domain, protocol Protocol) (*Socket, error) {
 	soc, err := C.nn_socket(C.int(domain), C.int(protocol))
-	if err != nil {
+	if soc == -1 {
 		return nil, nnError(err)
 	}
 

